app/src/main/java/org/mozilla/tv/firefox/architecture/FocusOnShowFragment.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.architecture
+
+import androidx.fragment.app.Fragment
+
+open class FocusOnShowFragment : Fragment() {
+    override fun onHiddenChanged(hidden: Boolean) {
+        if (!hidden) {
+            view?.requestFocus()
+        }
+        super.onHiddenChanged(hidden)
```

Comment:
I'm not super happy with how inflexible it is to have all of our fragments extend this class.  I wonder if we could find a better way to accomplish this via composition instead.  The best I've come up with so far is this, but there's probably a much nicer alternative that I haven't thought of.

```kotlin
class FocusOnShowDelegate {
    fun onHiddenChanged(fragment: Fragment, hidden: Boolean) {
        if (!hidden) {
            fragment.view?.requestFocus()
        }
    }
}

class SettingsFragment : Fragment() {
...
    override fun onHiddenChanged(hidden: Boolean) {
        FocusOnShowDelegate().onHiddenChanged(this, hidden)
        super.onHiddenChanged(hidden)
    }
...
}
```

Reply:
Yeah, that's at least a little better - I'll do that.